src/node/transaction.cpp

```diff
@@ -68,13 +69,15 @@ TransactionError BroadcastTransaction(NodeContext& node, const CTransactionRef t
             wtxid = mempool_tx->GetWitnessHash();
         } else {
             // Transaction is not already in the mempool.
-            if (max_tx_fee > 0) {
+            if (max_tx_fee > 0 || max_tx_feerate > CFeeRate(0)) {
                 // First, call ATMP with test_accept and check the fee. If ATMP
                 // fails here, return error immediately.
                 const MempoolAcceptResult result = node.chainman->ProcessTransaction(tx, /*test_accept=*/ true);
                 if (result.m_result_type != MempoolAcceptResult::ResultType::VALID) {
                     return HandleATMPError(result.m_state, err_string);
-                } else if (result.m_base_fees.value() > max_tx_fee) {
+                } else if (max_tx_fee > 0 && result.m_base_fees.value() > max_tx_fee) {
+                    return TransactionError::MAX_FEE_EXCEEDED;
+                } else if (max_tx_feerate > CFeeRate(0) && CFeeRate(result.m_base_fees.value(), result.m_vsize.value()) > max_tx_feerate) {
                     return TransactionError::MAX_FEE_EXCEEDED;
```

Comment:
The `MAX_FEERATE_EXCEEDED` error type was not introduced yet at this point.
 This is correct because at https://github.com/bitcoin/bitcoin/commit/eedbeccbdb274841fc2cf2c53931e58fbea50dfc, the error message indicates that the failure might be from `-maxtxfee` or `maxfeerate`.

And this is fixed in next commit 0ad2f2c32460fef9b8cd59e3920fcfd6602b02a4 after we introduced the `MAX_FEERATE_EXCEEDED` error type.

The options are;
- Introduce the error type first, then update` BroadcastTransaction` to support fee rate checks. This approach however, would result in misleading error messages in `sendrawtransaction`.
- Combine both changes in a single commit. 
- The current approach, update `BroadcastTransaction` to support checking fee rate then introduce the new error type which in my opinion, is the better approach.

Reply:
> The current approach, update BroadcastTransaction to support checking fee rate then introduce the new error type which in my opinion, is the better approach.

sounds good!